docs/pages/zero-trust-access/infrastructure-as-code/terraform-provider/env0.mdx

```diff
@@ -0,0 +1,225 @@
+environments, or in short-lived cloud VMs. In any of these cases, refer to a
+dedicated guide:
+
+- [Run the Terraform Provider on Terraform Cloud](./terraform-cloud.mdx)
+- [Run the Terraform Provider in CI or cloud VMs](./ci-or-cloud.mdx)
+- [Run the Terraform Provider locally](./local.mdx)
+
+## How it works
+
+When running the Teleport Terraform provider on env zero, you can use its
+built-in Machine ID support to dynamically authenticate to your Teleport cluster
+without any shared secrets. When run in this configuration, the Terraform
+provider proves its identity to the Teleport Auth Service using env zero's
+[OIDC Integration][oidc]. These are OIDC tokens provided to Terraform
+via an environment variable on the deployment runner, which the Teleport
+Terraform provider can use.
+
+While following this guide, you'll configure your Teleport cluster to accept
+join requests from env zero deployments and configure the provider to
+authenticate using the `env0` join method.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx edition="Teleport 18.4.0"!)
+
+- (!docs/pages/includes/tctl.mdx!)
+
+- [Terraform >= (=terraform.version=)+](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+  ```code
+  $ terraform version
+  # Terraform v(=terraform.version=)
+  ```
+
+  To import existing Teleport resources as Terraform resources, you must have
+  Terraform version `v1.5.0` or above.
+
+- An account and project on [env zero]
+
+- The Teleport Terraform provider, v18.4.0 or later
+
+## Step 1/5: Enable OIDC in env zero
+
+OIDC integrations must be enabled in your organization's credential settings. To
+do so:
+1. Navigate to your organization settings
+2. Select the "Policies" tab
+3. Under the "Third-Party Authentication" header, ensure "Enable OIDC during deployments" is checked
+4. If necessary, select "Save"
+
+For more information, refer to [env zero's documentation](https://docs.env0.com/docs/oidc-integrations#enabling-oidc-token-availability).
+
+## Step 2/5. Collect information
+
+To allow deployment workflows to authenticate to Teleport, you'll need to
+configure - at a minimum - the IDs of your organization and project. These
+values can be found in the env zero UI.
+
+To determine your organization ID, navigate to your organization's settings,
+select the "General" tab, and make a note of the value shown in the ID field. If
+desired, store your organization ID here: <Var name="example-organization-id" />
+
+Next, you'll also need to retrieve a project ID. Navigate to the settings for
+the desired project and make a note of the ID shown. If desired, you can insert
+it here: <Var name="example-project-id" />
+
+## Step 3/5. Configure env zero joining in Teleport
+
+To start, the Teleport Auth Service needs to be configured to accept join
+requests from env zero runs. We'll do this by creating a bot named `terraform`
+which the Teleport Terraform provider will use in a later step.
+
```

Comment:
I would make it explicit that the document below should be in `terraform_bot.yaml`.